btr-api/tests/unit/resources/test_submissions.py

```diff
@@ -147,6 +172,68 @@ def test_post_plots_pay_error(app, client, session, jwt, requests_mock):
             assert not created_submission.invoice_id
 
 
+def test_post_plots_frozen_entity(app, client, session, jwt, requests_mock):
+    """Assure no submission is created for a frozen entity."""
+    pay_api_mock = requests_mock.post(f"{app.config.get('PAYMENT_SVC_URL')}/payment-requests", exc=requests.exceptions.ConnectTimeout)
+
+    current_dir = os.path.dirname(__file__)
+    with open(
+        os.path.join(current_dir, '..', '..', 'mocks', 'significantIndividualsFiling', 'valid.json')
+    ) as file:
+        json_data = json.load(file)
+
+        mocked_entity_response = {"business" : {"adminFreeze" : True, "state" : "ACTIVE"}}
+        identifier = json_data['businessIdentifier']
+        legal_api_mock = requests_mock.get(
+            f"{app.config.get('LEGAL_SVC_URL')}/businesses/{identifier}",
+            json=mocked_entity_response
+        )
+
+        with nested_session(session):
+            rv = client.post('/plots',
+                             json=json_data,
+                             headers=create_header(jwt_manager=jwt,
+                                                   roles=['basic'],
+                                                   **{'Accept-Version': 'v1',
+                                                      'content-type': 'application/json',
+                                                      'Account-Id': 1}))
+            assert rv.status_code == HTTPStatus.BAD_REQUEST
+            assert rv.json.get('error') == ['The business is frozen']
+            assert legal_api_mock.called == True
+            assert pay_api_mock.called == False
+
+
+def test_post_plots_historical_entity(app, client, session, jwt, requests_mock):
```

Comment:
can you merge this test with above and set the `mocked_entity_response` and expected error via pytest params? It will remove a lot of duplication here